flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java

```diff
@@ -42,6 +45,16 @@ public void onReceive(final Context context, Intent intent) {
             Type type = new TypeToken<NotificationDetails>() {
             }.getType();
             NotificationDetails notificationDetails  = gson.fromJson(notificationDetailsJson, type);
+            if (notificationDetails.wakeScreen) {
+                PowerManager pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
```

Comment:
Could be moved closer to where it is used. (inside the if)

Reply:
Think it is fetched as close as it is used, ln 50-51 uses it ...`pm.isInteractive() : pm.isScreenOn();` then ln 53`pm.newWakeLock(`